### PR TITLE
[DOCS] Better debug text for errors generated by GridFieldDetailForm_ItemRequest

### PR DESCRIPTION
**Pull request by mattclegg**
See the commit log for details.

### CODE REVIEW

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 1]
```diff
@@ -111,7 +111,12 @@ public function Link($action = null)
     public function view($request)
     {
         if (!$this->record->canView()) {
-            $this->httpError(403);
+            $this->httpError(403, _t(
+                __CLASS__.'.ViewPermissionsFailure',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php]:
```suggestion
                __CLASS__ . '.ViewPermissionsFailure',
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 2]
```diff
@@ -111,7 +111,12 @@ public function Link($action = null)
...
+                'It seems you don\'t have the necessary permissions to view {ObjectTitle}',
+                '',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php]:
```suggestion
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 3]
```diff
@@ -189,8 +194,12 @@ public function ItemEditForm()
 
         if (!$this->record->canView()) {
             $controller = $this->getToplevelController();
-            // TODO More friendly error
-            return $controller->httpError(403);
+            return $controller->httpError(403, _t(
+                __CLASS__.'.ViewPermissionsFailure',
+                'It seems you don\'t have the necessary permissions to view {ObjectTitle}',
+                '',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php]:
```suggestion
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 4]
```diff
@@ -189,8 +194,12 @@ public function ItemEditForm()
 
         if (!$this->record->canView()) {
             $controller = $this->getToplevelController();
-            // TODO More friendly error
-            return $controller->httpError(403);
+            return $controller->httpError(403, _t(
+                __CLASS__.'.ViewPermissionsFailure',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php]:
```suggestion
                __CLASS__ . '.ViewPermissionsFailure',
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 5]
```diff
@@ -490,7 +499,13 @@ public function doSave($data, $form)
 
         // Check permission
         if (!$this->record->canEdit()) {
-            return $this->httpError(403);
+            $this->httpError(403, _t(
+                __CLASS__.'.EditPermissionsFailure',
+                'It seems you don\'t have the necessary permissions to edit {ObjectTitle}',
+                '',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php]:
```suggestion
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 6]
```diff
@@ -490,7 +499,13 @@ public function doSave($data, $form)
 
         // Check permission
         if (!$this->record->canEdit()) {
-            return $this->httpError(403);
+            $this->httpError(403, _t(
+                __CLASS__.'.EditPermissionsFailure',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php]:
```suggestion
                __CLASS__ . '.EditPermissionsFailure',
```

**Comment by dhensby** [comment]
Hey @robbieaverill - I've implemented your feedback on @mattclegg's behalf. I'm not sure about removing the empty parameter but I've done it anyway. I've also updated our linting rules so that string concatenation operators that aren't surrounded by spaces will throw from now on (PR reviews shouldn't be manual linting operations)

File: lang/en.yml [hunk 7]
```diff
@@ -111,6 +111,8 @@ en:
     NEW: 'Add new record'
     NEXT: 'Go to next record'
     PREVIOUS: 'Go to previous record'
+    EditPermissionsFailure: 'It seems you don''t have the necessary permissions to edit {ObjectTitle}'
+    ViewPermissionsFailure: 'It seems you don''t have the necessary permissions to view {ObjectTitle}'
```
Comment (robbieaverill) [inline lang/en.yml:115]:
```suggestion
    EditPermissionsFailure: 'It seems you don''t have the necessary permissions to edit "{ObjectTitle}"'
    ViewPermissionsFailure: 'It seems you don''t have the necessary permissions to view "{ObjectTitle}"'
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 8]
```diff
@@ -111,7 +111,11 @@ public function Link($action = null)
     public function view($request)
     {
         if (!$this->record->canView()) {
-            $this->httpError(403);
+            $this->httpError(403, _t(
+                __CLASS__ . '.ViewPermissionsFailure',
+                'It seems you don\'t have the necessary permissions to view {ObjectTitle}',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php:116]:
```suggestion
                'It seems you don\'t have the necessary permissions to view "{ObjectTitle}"',
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 9]
```diff
@@ -189,8 +193,11 @@ public function ItemEditForm()
 
         if (!$this->record->canView()) {
             $controller = $this->getToplevelController();
-            // TODO More friendly error
-            return $controller->httpError(403);
+            return $controller->httpError(403, _t(
+                __CLASS__ . '.ViewPermissionsFailure',
+                'It seems you don\'t have the necessary permissions to view {ObjectTitle}',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php:198]:
```suggestion
                'It seems you don\'t have the necessary permissions to view "{ObjectTitle}"',
```

File: src/Forms/GridField/GridFieldDetailForm_ItemRequest.php [hunk 10]
```diff
@@ -490,7 +497,12 @@ public function doSave($data, $form)
 
         // Check permission
         if (!$this->record->canEdit()) {
-            return $this->httpError(403);
+            $this->httpError(403, _t(
+                __CLASS__ . '.EditPermissionsFailure',
+                'It seems you don\'t have the necessary permissions to edit {ObjectTitle}',
```
Comment (robbieaverill) [inline src/Forms/GridField/GridFieldDetailForm_ItemRequest.php:502]:
```suggestion
                'It seems you don\'t have the necessary permissions to edit "{ObjectTitle}"',
```

**Comment by mattclegg** [comment]
Thanks for the review @robbieaverill & thanks for implementing the changes @dhensby 👍

**Pull request merged**